src/main.rs

```diff
@@ -1,35 +1,690 @@
+/// Though it is used by the `GiftWrap`, the `Patterns` enum can be used anywhere in Rust.
+#[derive(Debug, PartialEq)]
+pub enum Patterns {
+    Red,
+    Blue,
+    Green,
+    Paisley,
+    Polkadots,
+    Sparkles,
+    KraftPaper,
+    NewsPaper,
+    Cloth
+}
+
+/// The `GiftTag` struct represents a gift tag that could be included with a gift's wrapping. It is
+/// used by the `GiftWrap` struct to include a recipient, a sender, and a message. Though it is
+/// utilized by `GiftWrap`, this struct can be used anywhere in Rust.
+///
+/// `GiftTag` has the following parameters:
+/// * `recipient` which represents the recipient of the gift as a String.
+/// * `sender` which represents the sender of the gift as a String.
+/// * `message` which can be a message to be included with the gift as a String.
+///
+/// #Methods
+/// ##write(recipient, sender, message)
+/// You can write a new `GiftTag` with the [`GiftTag::write()`] method. Example:
+/// ```
+/// let tag = GiftTag::write(
+///      "Bob".to_string(),
+///      "Sally".to_string(),
+///      "Happy Cake Day!".to_string()
+/// );
+/// ```
+///
+/// ##read()
+/// You can read a `GiftTag` with the [`GiftTag::read()`] method. Example:
+/// ```
+/// assert_eq!(tag.read,
+/// "
+/// To: Bob,\
+/// From: Sally,\
+/// Message: Happy Cake Day!
+/// "
+/// );
+/// ```
+#[derive(Debug, PartialEq)]
+pub struct GiftTag {
+    recipient: String,
+    sender: String,
+    message: String
+}
+
+impl GiftTag {
+    /// The `write` method accepts three arguments as Strings (a recipient, a sender, and a message)
+    /// and returns a `GiftTag`.
+    ///
+    /// #Arguments
+    /// * `recipient` - Accepts a String that represents a gift's recipient (the person receiving the
+    /// gift).
+    /// * `sender` - Accepts a String that represents a gift's sender (the person who sent the gift).
+    /// * `message` - Accepts a string that represents a message to the recipient from the sender to
+    /// be included with the gift.
+    ///
+    /// #Returns
+    /// Returns a `GiftTag`.
+    ///
+    /// #Example
+    /// ```
+    /// let tag = GiftTag::write(
+    ///      "Bob".to_string(),
+    ///      "Sally".to_string(),
+    ///      "Happy Cake Day!".to_string()
+    /// );
+    /// ```
+    pub fn write(recipient: String, sender: String, message: String) -> GiftTag {
+        GiftTag {
+            recipient,
+            sender,
+            message
+        }
+    }
+
+    /// The `read()` method takes a `GiftTag` as `self` and returns a formatted String representing
+    /// the contents of the `GiftTag`.
+    ///
+    /// #Arguments
+    /// * `self` only.
+    ///
+    /// #Returns
+    /// Returns a pre-formatted String.
+    ///
+    /// #Example
+    /// ```
+    /// assert_eq!(tag.read,
+    /// "
+    /// To: Bob,\
+    /// From: Sally,\
+    /// Message: Happy Cake Day!
+    /// "
+    /// );
+    /// ```
+    pub fn read(self) -> String {
```

Comment:
Similar to the fill function above, instead of implementing this as a custom function, the https://doc.rust-lang.org/std/fmt/trait.Display.html trait is the standard way to handle visually displaying the tag. Then instead of calling read_tag, one could just use the format! macro to display the value to a string.

```rust
let tag = ...;

// Without Display trait implemented (inserting a value with the String type to the first {})
let nice_looking_formatted_tag = format!("The tag says: {}", tag.read_tag());
// With Display trait implemented (inserting a value with the Tag type to the first {}, and allow the formatter/implmentation worry about how that ends up printing out)
let nice_looking_formatted_tag = format!("The tag says: {}", tag);
```

This will also let the tag be displayed in any other context with the format! macro as well

Reply:
I've added the fmt trait to GiftTag and fmt now does the heavy lifting of formatting the string, instead of doing it in the read function. However, I've kept the read function because I think it's a nice way to introduce this concept from first "reading" the tag using the function, then exploring how the "read" function works by using the implemented fmt trait.

What do you think of this move?

Changes made in commit https://github.com/hypnoseal/giftbox/commit/30764163235a6df7e5d23492021b57a8519c4a53